controllers/harbor/provision.go

```diff
@@ -0,0 +1,307 @@
+package harbor
+
+import (
+	goharborv1 "github.com/goharbor/harbor-cluster-operator/api/v1"
+	"github.com/goharbor/harbor-cluster-operator/lcm"
+	"github.com/goharbor/harbor-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Provision will create a new Harbor CR.
+func (harbor *HarborReconciler) Provision() (*lcm.CRStatus, error) {
+	harborCR := harbor.newHarborCR()
+	err := harbor.Create(harborCR)
+	if err != nil {
+		return harborCRNotReadyStatus("", ""), err
+	}
+	return harborCRUnknownStatus(), err
+}
+
+// newHarborCR will create a new Harbor CR controlled by harbor-operator
+func (harbor *HarborReconciler) newHarborCR() *v1alpha1.Harbor {
+	namespacedName := harbor.getHarborCRNamespacedName()
+
+	return &v1alpha1.Harbor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      namespacedName.Name,
+			Namespace: namespacedName.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(harbor.HarborCluster, goharborv1.HarborClusterGVK),
+			},
+		},
+		Spec: v1alpha1.HarborSpec{
+			HarborVersion: harbor.HarborCluster.Spec.Version,
+			PublicURL:     harbor.HarborCluster.Spec.PublicURL,
+			TLSSecretName: harbor.HarborCluster.Spec.TLSSecret,
+			Components: v1alpha1.HarborComponents{
+				Core:        harbor.newCoreComponentIfNecessary(),
```

Comment:
Core, portal, registry, and job service are mandatory components, right? It seems they do not match `IfNecessary` case.